torch/_torch_docs.py

```diff
@@ -8038,6 +8038,11 @@ def merge_dicts(*dicts):
     possible to satisfy the definition of a modulo operation with complex numbers.
     See :func:`torch.fmod` for how division by zero is handled.
 
+.. note::
+    This is different to `math.remainder <https://docs.python.org/dev/library/math.html#math.remainder>`_
```

Comment:
"This is different to" -> "This is different from"

Can we explain HOW it's different succinctly?  Like, what do std::remainder and math.remainder do differently? Also, why is PyTorch divergent from them? We typically try to follow what C++ and Python do. 

What's even more interesting is that the Python Array API defines [remainder](https://data-apis.org/array-api/latest/API_specification/elementwise_functions.html?highlight=remainder). That definition may be underspecified. cc @rgommers 

So I think there are still interesting questions of fact to be resolved with torch.remainder.

Reply:
> Can we explain HOW it's different succinctly? Like, what do std::remainder and math.remainder do differently? 

`std::remainder` and `math.remainder` try to implement the remainder operation defined by IEEE 754 standard. Here is the relevant excerpt from Section 5.1 of IEEE 754-1985 Standard:

> 5.1 Arithmetic
> When y != 0, the remainder r = x REM y is defined regardless of the rounding mode by the mathematical relation r = x - y
> ∗ n, where n is the integer nearest the exact value x/y; whenever |n-x/y| = 1/2, then n is even. Thus, the remainder is always exact. If r = 0, its sign shall be that of x. Precision control (4.3) shall not apply to the remainder operation.

While NumPy tries to implement the "%" operator of Python (modulus operator), here is what Python defines it as (link: https://docs.python.org/3.7/reference/expressions.html#binary-arithmetic-operations):

> The % (modulo) operator yields the remainder from the division of the first argument by the second. The numeric 
> arguments are first converted to a common type. A zero right argument raises the ZeroDivisionError exception. The 
> arguments may be floating point numbers, e.g., 3.14%0.7 equals 0.34 (since 3.14 equals 4*0.7 + 0.34.) The modulo 
> operator always yields a result with the same sign as its second operand (or zero); the absolute value of the result is 
> strictly smaller than the absolute value of the second operand.

The difference that I can understand is:

| Python (NumPy/PyTorch) | IEEERemainder (C++ std::remainder, math.remainder) |
| ------------------------- | ---------------------  |
| Same sign as the second operand | No same sign guaranteed except when dividing by zero (same sign as first operand or dividend) |
| Rounding used: floor | Rounding used: closest integer |

(I'll update this comment with the formulation as well)

Reply:
I think the intent there is to do the same as Python, NumPy and PyTorch. I'll add it to the list of things that need clarification.

Reply:
Being clearer about the difference would be great. Additionally, I guess we should implement torch.mod and make torch.remainder an alias to it? NumPy has fmod, mod, and remainder. I think PyTorch just has fmod and remainder currently.

The documentation of torch.fmod will also have to be updated.

Reply:
I have updated the doc for `torch.remainder`, here is the rendered version: https://14874396-65600975-gh.circle-artifacts.com/0/docs/generated/torch.remainder.html#torch.remainder. PTAL, @mruberry whenever you find time.

I agree that PyTorch should have a `torch.mod` function and making `remainder` alias to it. Please correct me if I'm wrong, implementation wise we don't need to change anything (w.r.t kernels etc.) - all we should do is add an entry to `torch.mod` and alias it to `remainder`? Or if we want to rename everything from `remainder` to `mod` and then add a `torch.remainder` entry which aliases to `torch.mod`? Just confirming before it's implemented, thanks!

> The documentation of torch.fmod will also have to be updated.

AFAIK, `torch.fmod` is consistent with `std::fmod` since it itself calls `std::fmod` (for floating types): https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L813.

I'm not sure why does it need any updates? Is there anything I'm missing? The current documentation looks good to me: https://14874396-65600975-gh.circle-artifacts.com/0/docs/generated/torch.fmod.html (we updated recently in the OpInfo PR of fmod and remainder, to add difference between torch.remainder and torch.fmod).

Reply:
You are correct we would just add mod and, internally, mod could be an alias to remainder (or the other way around)

I think you're right that torch.fmod doesn't need any updates

Reply:
Thanks @mruberry for the clarification.